app/upgrades/v16/utils.go

```diff
@@ -0,0 +1,140 @@
+	} else if utils.IsTestnet(ctx.ChainID()) {
+		wrappedContractAddr = common.HexToAddress(WEVMOSContractTestnet)
+	}
+
+	// iterate over all the accounts and convert the tokens to native coins
+	accountKeeper.IterateAccounts(ctx, func(account authtypes.AccountI) (stop bool) {
+		// FIXME: the accounts created with the keyring are not EthAccounts? Is asserting for BaseAccount good enough?
+		baseAccount, ok := account.(*authtypes.BaseAccount)
+		// baseAccount, ok := account.(*evmostypes.EthAccount)
+		if !ok {
+			return false
+		}
+
+		cosmosAddress := baseAccount.GetAddress()
+		ethAddress := common.BytesToAddress(cosmosAddress.Bytes())
+		ethHexAddr := ethAddress.String()
+
+		balance, res, err := WithdrawWEVMOS(ctx, ethAddress, wrappedContractAddr, erc20Keeper)
+		if err != nil {
+			logger.Debug(
+				"failed to withdraw WEVMOS",
+				"account", ethHexAddr,
+				"balance", balance.String(),
+				"error", err.Error(),
+			)
+		} else if res != nil && res.VmError != "" {
+			logger.Debug(
+				"withdraw WEVMOS reverted",
+				"account", ethHexAddr,
+				"balance", balance.String(),
+				"vm-error", res.VmError,
+			)
+		}
+
+		erc20Keeper.IterateTokenPairs(ctx, func(tokenPair erc20types.TokenPair) bool {
+			if !tokenPair.IsNativeCoin() {
+				return false
+			}
+
+			contract := tokenPair.GetERC20Contract()
+
+			if err := ConvertERC20Token(ctx, ethAddress, contract, cosmosAddress, erc20Keeper); err != nil {
+				logger.Debug(
+					"failed to convert ERC20 to native Coin",
+					"account", ethHexAddr,
+					"erc20", contract.String(),
+					"balance", balance.String(),
+					"error", err.Error(),
+				)
+			}
+
+			return false
+		})
+
+		return false
+	})
+
+	erc20ModuleAccountAddress := authtypes.NewModuleAddress(erc20types.ModuleName)
+	balances := bankKeeper.GetAllBalances(ctx, erc20ModuleAccountAddress)
+	if balances.IsZero() {
+		return nil
+	}
+
+	// burn all the coins left in the module account
+	return bankKeeper.BurnCoins(ctx, erc20types.ModuleName, balances)
+}
+
+func WithdrawWEVMOS(ctx sdk.Context, from, wevmosContract common.Address, erc20Keeper erc20keeper.Keeper) (*big.Int, *evmtypes.MsgEthereumTxResponse, error) {
+	balance := erc20Keeper.BalanceOf(ctx, contracts.ERC20MinterBurnerDecimalsContract.ABI, wevmosContract, from)
+	if balance == nil {
+		return common.Big0, nil, fmt.Errorf("failed to get WEVMOS balance for %s", from.String())
+	}
+
+	// only execute the withdrawal if balance is positive
+	if balance.Cmp(common.Big0) < 1 {
+		return common.Big0, nil, nil
+	}
+
+	// call withdraw method from the account
+	var data []byte
+	res, err := erc20Keeper.CallEVMWithData(ctx, from, &wevmosContract, data, true)
+	return balance, res, err
+}
+
+func ConvertERC20Token(ctx sdk.Context, from, contract common.Address, receiver sdk.AccAddress, erc20Keeper erc20keeper.Keeper) error {
+	balance := erc20Keeper.BalanceOf(ctx, contracts.ERC20MinterBurnerDecimalsContract.ABI, contract, from)
+	if balance.Cmp(common.Big0) <= 0 {
```

Comment:
ditto
```suggestion
	if balance == nil || balance.Sign() < 1 {
```

Reply:
same as above, I think separating the cases for `balance == nil` and the non-positive amounts makes sense because in one case there was an error raised